### PR TITLE
feat(MOPLAT-262): render show Installs mansory list

### PR DESCRIPTION
**Pull request by patrinoua**
This PR resolves [MOPLAT-262]

## Done
- Show Install shots List
- Navigate to fullscreen image

Screen Shot 2022-07-21 at 13 25 02

Screen Shot 2022-07-21 at 18 20 25

### PR Checklist

- [ ] I tested my changes on **iOS** and **Android**.
- [ ] I added screenshots or videos to illustrate my changes.
- [ ] I added Tests and Stories for my changes.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

Need help with something? Have a look at our [docs], or get in touch with us.

[docs]: /docs/README.md

[MOPLAT-262]: https://artsyproduct.atlassian.net/browse/MOPLAT-262?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by pvinis — APPROVED** [review]

File: src/app/sharedUI/screens/InstallImage/InstallImage.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  const insets = useSafeAreaInsets()
+  const navigation = useNavigation<NavigationProp<HomeTabsScreens>>()
+
+  return (
+    <Flex flex={1} pt={insets.top} mt={2}>
+      <Flex px={2}>
+        <Touchable
+          onPress={() => {
+            navigation.goBack()
+          }}
```
Comment (rajsam003) [inline src/app/sharedUI/screens/InstallImage/InstallImage.tsx:20]: onPress={() => navigation.goBack()}

Comment (rajsam003) [inline src/app/sharedUI/screens/InstallImage/InstallImage.tsx:20]:
```suggestion
          onPress={() =>  navigation.goBack()}
```

File: src/app/sharedUI/items/ArtworkImageGridItem.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  return (
+    <Touchable
+      onPress={() =>
+        navigation.navigate("InstallImage", {
+          url,
+        })
+      }
+    >
+      <Flex mb={4} pl={2}>
+        <Image source={{ uri: url }} style={{ aspectRatio: 1 }} />
```
Comment (rajsam003) [inline src/app/sharedUI/items/ArtworkImageGridItem.tsx:22]: **non-blocking** 
```suggestion
        <Image source={{ url }} style={{ aspectRatio: 1 }} />
```

File: src/app/screens/HomeTabs/Shows/ShowTabs/ShowInstalls/ShowInstalls.tsx [hunk 3]
```diff
@@ -1,11 +1,41 @@
...
 
-export const ShowInstalls = () => {
   return (
-    <TabsFlatList
-      data={[1, 2, 3, 4, 5]}
-      renderItem={({ item }) => <Text>Show installs {item}</Text>}
-    />
+    <TabsScrollView>
+      <MasonryList
+        testID="show-artwork-list"
```
Comment (rajsam003) [inline src/app/screens/HomeTabs/Shows/ShowTabs/ShowInstalls/ShowInstalls.tsx:15]: **thought**: As we have used relay in this file, so maybe we can add test cases.

File: src/app/sharedUI/items/ArtworkImageGridItem.tsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import { Flex, Touchable } from "palette"
+import { Image } from "react-native"
+import { NavigationProp, useNavigation } from "@react-navigation/native"
+import { HomeTabsScreens } from "app/routes/HomeTabsNavigationStack"
+
+interface ArtworkImageGridItemProps {
+  url: string
+}
+
+export const ArtworkImageGridItem: React.FC<ArtworkImageGridItemProps> = ({ url }) => {
```
Comment (rajsam003) [inline src/app/sharedUI/items/ArtworkImageGridItem.tsx:10]: **non-blocking**
```suggestion
export const ArtworkImageGridItem= ({ url }:{url: string}) => {
```

Comment (rajsam003) [reply src/app/sharedUI/items/ArtworkImageGridItem.tsx:10]: As it gets just one prop

**Comment by rajsam003** [comment]
Good work @patrinoua 
Left some comments and suggestions, but feel free to ignore them if you don't agree with them

File: src/app/screens/HomeTabs/Shows/ShowTabs/ShowInstalls/ShowInstalls.tsx [hunk 5]
```diff
@@ -1,11 +1,41 @@
...
+        contentContainerStyle={{
+          marginTop: 20,
+          paddingRight: 20,
```
Comment (rajsam003) [inline src/app/screens/HomeTabs/Shows/ShowTabs/ShowInstalls/ShowInstalls.tsx:18]: **non-blocking**
```suggestion
          marginTop: space(2),
          paddingRight: space(2),
```
We can use useSpace() here. 
I just created PR for it. 
https://github.com/artsy/energy/pull/107

File: src/app/sharedUI/items/ArtworkImageGridItem.tsx (hunk 2 above)
Comment (patrinoua) [reply src/app/sharedUI/items/ArtworkImageGridItem.tsx:22]: Image expects an **uri**: _A string representing the resource identifier for the image, which could be an http address, a local file path, or the name of a static image resource._

https://reactnative.dev/docs/image
https://reactnative.dev/docs/image#imagesource

File: src/app/screens/HomeTabs/Shows/ShowTabs/ShowInstalls/ShowInstalls.tsx [hunk 6]
```diff
@@ -1,11 +1,41 @@
...
+        }}
+        numColumns={2}
+        data={installs as unknown[]}
```
Comment (rajsam003) [inline src/app/screens/HomeTabs/Shows/ShowTabs/ShowInstalls/ShowInstalls.tsx:21]: @pvinis Could you help us with this typing?

File: patches/@react-native-seoul+masonry-list+1.3.0.patch [hunk 7]
```diff
@@ -0,0 +1,13 @@
+diff --git a/node_modules/@react-native-seoul/masonry-list/lib/index.d.ts b/node_modules/@react-native-seoul/masonry-list/lib/index.d.ts
+index 3458b95..4347eb9 100644
+--- a/node_modules/@react-native-seoul/masonry-list/lib/index.d.ts
++++ b/node_modules/@react-native-seoul/masonry-list/lib/index.d.ts
+@@ -8,7 +8,7 @@ interface Props<T> extends Omit<ScrollViewProps, 'refreshControl'> {
+     onEndReached?: () => void;
+     onEndReachedThreshold?: number;
+     style?: StyleProp<ViewStyle>;
+-    data: T[];
++    data: readonly T[];
```
Comment (rajsam003) [inline patches/@react-native-seoul+masonry-list+1.3.0.patch:10]: 👀

**Pull request merged**